pkg/syncer/cnsoperator/util/util.go

```diff
@@ -120,11 +130,47 @@ func GetTKGVMIP(ctx context.Context, vmOperatorClient client.Client, dc dynamic.
 		if !exists {
 			return "", fmt.Errorf("failed to get SNAT IP annotation from VirtualMachine %s/%s", vmNamespace, vmName)
 		}
-	} else {
+	} else if network_provider_type == VDSNetworkProvider {
 		ip = virtualMachineInstance.Status.VmIp
 		if ip == "" {
 			return "", fmt.Errorf("vm.Status.VmIp is not populated for %s/%s", vmNamespace, vmName)
 		}
+	} else {
+		vpcName := vmNamespace
```

Comment:
if capability is not set, do we want to keep this code or return unsupported error?

Reply:
Updated getVMExternalIP to only check for nsx_vpc when capability is enabled.